flent/scripts/ethtool_iterate.sh

```diff
@@ -0,0 +1,182 @@
+#!/bin/bash
+# SPDX-License-Identifier: GPL-3.0
+# SPDX-FileContributor: Freysteinn Alfredsson <freysteinn@freysteinn.com>
+# SPDX-FileType: SOURCE
+
+count=10
+interval=0.1
+host=localhost
+exact_values=0
+
+while getopts "c:I:H:d:f:e" opt; do
+    case $opt in
+        c) count=$OPTARG ;;
+        I) interval=$OPTARG ;;
+        H) host=$OPTARG ;;
+        d) devices=$OPTARG ;;
+        f) fields=$OPTARG ;;
+        e) exact_values=1 ;;
+    esac
+done
+
+command_string=$(cat <<EOF
+awk -v COUNT="$count" \
+    -v INTERVAL="$interval" \
+    -v DEVICES="$devices" \
+    -v FIELDS="$fields" \
+    -v EXACT_VALUES="$exact_values" ' \
```

Comment:
This script is doing way too much work on the captured host. The data gathering scripts are deliberately simple, so they can work on constrained devices, and so the data parting and smarts are implemented in Python in Flent. Also, looping through all the interfaces every iteration can take way too long on a constrained device, making the data points "late" relative to the polling interval. This is the reason the other runners spawn a separate runner instance per interface, for instance; we should do the same here.

So basically, I would change this script to be something like:
```sh
patterns=""
for f in fields; do
    patterns="-e '$f' $patterns"
done

command_string=$(cat <<EOF
for i in \$(seq $count); do
    date '+Time: %s.%N';
    ethtool -S $interface | grep -w $patterns
    echo "---";
    sleep $interval || exit 1;
done
EOF
)
```

and then do all the field parsing in the Flent code; and simply spawn a helper for each interface, like the qdisc and wifi stats do.